app/src/main/java/org/mozilla/fenix/home/pocket/PocketStoriesViewHolder.kt

```diff
@@ -52,94 +54,61 @@ class PocketStoriesViewHolder(
 
     @Composable
     override fun Content() {
-        PocketStories(
-            store,
-            interactor::onStoriesShown,
-            interactor::onStoryClicked,
-            interactor::onCategoryClicked,
-            interactor::onDiscoverMoreClicked,
-            interactor::onLearnMoreClicked,
-            with(composeView.resources) {
-                getDimensionPixelSize(R.dimen.home_item_horizontal_margin) / displayMetrics.density
-            }
-        )
-    }
-}
-
-@Composable
-@Suppress("LongParameterList")
-fun PocketStories(
-    store: HomeFragmentStore,
-    onStoriesShown: (List<PocketRecommendedStory>) -> Unit,
-    onStoryClicked: (PocketRecommendedStory, Pair<Int, Int>) -> Unit,
-    onCategoryClicked: (PocketRecommendedStoriesCategory) -> Unit,
-    onDiscoverMoreClicked: (String) -> Unit,
-    onLearnMoreClicked: (String) -> Unit,
-    @Dimension horizontalPadding: Float = 0f
-) {
-    val stories = store
-        .observeAsComposableState { state -> state.pocketStories }.value
+        val horizontalPadding = dimensionResource(R.dimen.home_item_horizontal_margin)
 
-    val categories = store
-        .observeAsComposableState { state -> state.pocketStoriesCategories }.value
+        val stories = store
+            .observeAsComposableState { state -> state.pocketStories }.value
 
-    val categoriesSelections = store
-        .observeAsComposableState { state -> state.pocketStoriesCategoriesSelections }.value
+        LaunchedEffect(stories) {
+            // We should report back when a certain story is actually being displayed.
+            // Cannot do it reliably so for now we'll just mass report everything as being displayed.
+            stories?.let {
+                interactor::onStoriesShown
```

Comment:
@Mugurell  i think there is a problem here! you forgot to invoke this method. instead used a method reference